judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/transfer/MappedTransferAggregationTest.java

```diff
@@ -0,0 +1,410 @@
+    TransferEDao transferEDao;
+
+    @Inject
+    EntityFDao entityFDao;
+
+    @Inject
+    TransferFDao transferFDao;
+
+
+    @Inject
+    EntityGDao entityGDao;
+
+    @Inject
+    TransferGDao transferGDao;
+
+
+    @Override
+    public Module getModelDaoModule() {
+        return new MappedTransferAggregationDaoModules();
+    }
+
+    @Override
+    public String getModelName() {
+        return "MappedTransferAggregation";
+    }
+
+    /**
+     * The test checks the single aggregation work well on transfer object.
+     *
+     * @prerequisites The test must start and finish on the same day. Therefore, don't run this test close to midnight.
+     *
+     * @type Behaviour
+     *
+     * @others Implement this test case in the *judo-runtime-core-jsl-itest* module.
+     *
+     * @jslModel MappedTransferAggregation.jsl
+     *
+     * @positiveRequirements REQ-ENT-005
+     *
+     */
+    @Test
+    @TestCase("SingleAggregationOnTransfer")
+    @Requirement(reqs = {
+            "REQ-MDL-001",
+            "REQ-MDL-002",
+            "REQ-MDL-003",
+            "REQ-TYPE-001",
+            "REQ-TYPE-004",
+            "REQ-ENT-001",
+            "REQ-ENT-002",
+            "REQ-ENT-004",
+            "REQ-ENT-005",
+            "REQ-SRV-001"
+    })
+    public void testSingleAggregationOnTransfer() {
+
+        TransferB transferB = transferBDao.create(TransferB.builder().withNameB("B1").build());
+
+        assertEquals(1, transferBDao.query().execute().size());
+
+        TransferA transferA = transferADao.create(TransferA.builder()
+                .withSingleEntityB(transferB)
+                .build()
+        );
+
+        // TODO JNG-4317 Composition should copy the composition element or not allow to bind an existed element.
+//        assertEquals(2,transferBDao.query().execute().size());
+//        assertEquals(2,entityBDao.query().execute().size());
+
+        // Check transferA contains transferB
+        assertEquals(transferB.identifier(),transferA.getSingleEntityB().orElseThrow().identifier());
+        assertEquals(
+                entityBDao.getById(transferB.adaptTo(EntityBIdentifier.class)).orElseThrow().identifier(),
+                entityADao.getById(transferA.adaptTo(EntityAIdentifier.class)).orElseThrow().getSingleEntityB().orElseThrow().identifier()
+        );
+
+        // Check transferA cannot bind a different B element
+        transferA.setSingleEntityB(transferBDao.create(TransferB.builder().withNameB("B2").build()));
+        IllegalStateException thrown = assertThrows(
+                IllegalStateException.class,
+                () -> transferADao.update(transferA)
+        );
+        assertTrue(thrown.getMessage().contains("Identifier cannot be different on containment reference element"));
+        assertTrue(thrown.getMessage().contains("#singleEntityB"));
+
+        // Check transferA can set to null
+        transferA.setSingleEntityB(null);
+        transferADao.update(transferA);
```

Comment:
After calling update, transferA java object must be reassigned, because update returns a "new" object of transferA.

I recommend reassigning after updates that have failed as expected.

Reply:
[696e1db](https://github.com/BlackBeltTechnology/judo-runtime-core-jsl/pull/182/commits/696e1db01ddcd628196a4fd71753d1460bed4e2d)